### PR TITLE
Add "onDownloadProgress" callback for chromium download

### PR DESCRIPTION
**Pull request by Wizzel1**
I thought it might be useful to give the user some information about the download progress of chromium.

### CODE REVIEW

**Comment by xvrh** [comment]
Thanks for the proposal.
Can you do it using `package:http`? 

I really want to limit the number of third party dependencies. `package:http` is already there and is maintained by the Dart team.

**Comment by Wizzel1** [comment]
@xvrh sure, should work now

File: lib/src/downloader.dart [hunk 1]
```diff
@@ -58,14 +73,26 @@ Future<RevisionInfo> downloadChrome({int? revision, String? cachePath}) async {
...
-  await response.stream.pipe(ouputFile.openWrite());
-  client.close();
+Future _downloadFile(
+  String url,
+  String output,
+  Function(int, int?)? onRecieveProgress,
+) async {
+  final client = http.Client();
+  final response = await client.send(http.Request('get', Uri.parse(url)));
+  final totalBytes = response.contentLength;
```
Comment (Wizzel1) [inline lib/src/downloader.dart:83]: I wasn't sure how to handle the case if the the size of the request is not known in advance. 
`totalBytes' will be null in that case.

Comment (xvrh) [reply lib/src/downloader.dart]: I think defaulting to zero is good enough. I pushed a commit on your branch with that.

Comment (Wizzel1) [reply lib/src/downloader.dart]: Okay, sounds good!

**Comment by xvrh** [comment]
Thanks for the contribution!

**Pull request merged**